File: java_app/src/main/java/com/petnagy/java/example/redux/LogMiddleware.java
package com.petnagy.java.example.redux;

import com.petnagy.koredux.Action;
import com.petnagy.koredux.DispatchFunction;
import com.petnagy.koredux.Middleware;
import com.petnagy.koredux.Store;

import org.jetbrains.annotations.NotNull;

import timber.log.Timber;

public class LogMiddleware implements Middleware<AppState> {
    @Override
    public void invoke(@NotNull Store<AppState> store, @NotNull Action action, @NotNull DispatchFunction next) {
        Timber.d("Action: -> %s", action.getClass().getSimpleName());
        next.dispatch(action);
    }
}

File: app/src/main/java/com/petnagy/app/redux/LogMiddleware.kt
package com.petnagy.app.redux

import com.petnagy.koredux.Action
import com.petnagy.koredux.DispatchFunction
import com.petnagy.koredux.Middleware
import com.petnagy.koredux.Store
import timber.log.Timber

class LogMiddleware: Middleware<AppState> {
    override fun invoke(store: Store<AppState>, action: Action, next: DispatchFunction) {
        var log = "Action: -> " + action::class.java.simpleName
        Timber.d(log)
        next.dispatch(action)
    }
}